freva-rest/README.md

```diff
@@ -45,33 +45,6 @@ The following default values are available on start up:
 
 ```console
 freva-rest-server --help
```

Comment:
would it be too much to show the help command here?
```bash
Usage: freva-rest-server [-h] [--port PORT] [--config CONFIG] [--api-url API_URL] [--debug] [--mongo-host MONGO_HOST] [--mongo-user MONGO_USER]
                         [--mongo-password MONGO_PASSWORD] [--mongo-db MONGO_DB] [--solr-host SOLR_HOST] [--solr-core SOLR_CORE] [--cache-exp CACHE_EXP]
                         [--redis-host REDIS_HOST] [--redis-ssl-certfile REDIS_SSL_CERTFILE] [--redis-ssl-keyfile REDIS_SSL_KEYFILE]
                         [--redis-password REDIS_PASSWORD] [--redis-user REDIS_USER] [--oidc-discovery-url OIDC_DISCOVERY_URL] [--oidc-client-id OIDC_CLIENT_ID]
                         [--oidc-client-secret OIDC_CLIENT_SECRET] [--dev] [--n-workers N_WORKERS] [--redis-ssl-certdir REDIS_SSL_CERTDIR]
                         [--services {databrowser,zarr-stream} [{databrowser,zarr-stream} ...]]

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃                                                    Command line interface (cli) for running the rest server.                                                     ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

                                                                            Configuring

There are three fundamental different options to configure the service.

 1 via command line arguments
 2 via environment variables.
 3 via the config .toml file.

Note, that the order here is important. First, given command line arguments are evaluated, if present any of the below given environment variables are evaluated,
finally the api config file is evaluated. The following environment variables are evaluated:

 • DEBUG: Start server in debug mode (1), (default: 0 -> no debug).
 • API_PORT: the port the rest service should be running on (default 8080).
 • API_WORKER: the number of multi-process work serving the API (default: 8).
 • API_SOLR_HOST: host name of the solr server, host name and port should be separated by a :, for example localhost:8983
 • API_SOLR_CORE : name of the solr core that contains datasets with multiple versions
 • API_MONGO_HOST: host name of the mongodb server, where query statistics are stored. Host name and port should separated by a :, for example localhost:27017
 • API_MONGO_USER: user name for the mongodb.
 • API_MONGO_PASSWORD: password to log on to the mongodb.
 • API_MONGO_DB: database name of the mongodb instance.
 • API_URL: url of the machine that runs of the rest api
 • API_CACHE_EXP: expiry time in seconds of the cached data
 • API_REDIS_HOST: Host and port of the redis cache Host name and port should separated by a :, for example localhost:5672
 • API_REDIS_PASS: Password for the redis connection.
 • API_REDIS_USER: Username for the redis connection.
 • API_REDIS_SSL_CERTFILE: Path to the TSL certificate file used to encrypt the redis connection.
 • API_REDIS_SSL_KEYFILE: Path to the TSL key file used to encrypt the redis connection.
 • API_OIDC_URL: Discovery of the open connect id service.
 • API_OIDC_CLIENT_ID: Name of the client (app) that is used to create the access tokens, defaults to freva
 • API_OIDC_CLIENT_SECRET: You can set a client secret, if you have
 • API_SERVICES:  The services the api should serve.

📝  You can override the path to the default config file using the API_CONFIG environment variable. The default location of this config file is
/opt/databrowser/api_config.toml.

Options:
  -h, --help            show this help message and exit
  --port, -p PORT       Set the port the server should be running on. (default: 7777)
  --config CONFIG       Path to a .toml file holding the APIconfiguration (default: /Users/mo/dev/20241112/freva-nextgen/freva-rest/src/freva_rest/api_config.toml)
  --api-url API_URL     The url of the host serving the API. (default: None)
  --debug               Turn on debug mode (default: False)
  --mongo-host MONGO_HOST
                        Set the <HOSTNAME>:<PORT> to the MongoDB service. (default: None)
  --mongo-user MONGO_USER
                        The mongoDB user name to log on to the mongoDB. (default: None)
  --mongo-password MONGO_PASSWORD
                        The MongoDb password to log on to the mongoDB. (default: None)
  --mongo-db MONGO_DB   Name of the Mongo database that is used. (default: None)
  --solr-host SOLR_HOST
                        Set the <HOSTNAME>:<PORT> to the Solr service. (default: None)
  --solr-core SOLR_CORE
                        Set the name of the core for the search index. (default: None)
  --cache-exp CACHE_EXP
                        The expiration time in secof the data loading cache. (default: None)
  --redis-host REDIS_HOST
                        Url of the redis cache. (default: None)
  --redis-ssl-certfile REDIS_SSL_CERTFILE
                        Path to the publiccertfile to makeconnections to thecache (default: None)
  --redis-ssl-keyfile REDIS_SSL_KEYFILE
                        Path to the privatkey file to makeconnections to thecache (default: None)
  --redis-password REDIS_PASSWORD
                        Password for redis connections. (default: None)
  --redis-user REDIS_USER
                        Username for redis connections. (default: None)
  --oidc-discovery-url OIDC_DISCOVERY_URL
                        OpenID connect discovery url. (default: None)
  --oidc-client-id OIDC_CLIENT_ID
                        The OIDC client id used for authentication. (default: None)
  --oidc-client-secret OIDC_CLIENT_SECRET
                        The OIDC client secret, if any, used for authentication. (default: None)
  --dev                 Enable development mode (default: False)
  --n-workers, -w N_WORKERS
                        Number of parallel processes. (default: 8)
  --redis-ssl-certdir REDIS_SSL_CERTDIR
                        The directory where the certficates are stored. This can be used to instead of setting the cert, and key files. (default: None)
  --services, -s {databrowser,zarr-stream} [{databrowser,zarr-stream} ...]
                        The services the API should serve (default: [''])
```

Reply:
In my next PR I want to introduce a very simple API where users (admins) can add (code) their search engine definitions. Those settings will then be displayed by the cli. This is the reason why I went from typer to good ole argparse, because it's more flexible. Having said that. The specifics of the help messages might change if another backend in implemented. Hence I figured to mention only the commands that are available after installation.


I will slightly change the README.